demos/basics/loading-saving-neo4j.ipynb

```diff
@@ -0,0 +1,2038 @@
+{
```

Comment:
<blockquote>One way sort of a subgraph someone might be interested is one where the nodes and/or edges satisfy certain criteria.</blockquote><p>"One type of subgraph someone might be interested..."?</p><p> </p><p>Also in general, this <code>Subgraphs</code>  sections feels more like a cypher tutorial to me, rather than anything <code>stellargraph</code>  specific - I understand that it comes from us believing it would be a common use-case since you probably wouldn't load ALL your neo4j data into memory, but the exact way the user should reduce their graph is probably going to be very specific to their data. Thoughts?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1184/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Reply:
This ~whole notebook is a cypher tutorial, since it's actually not doing anything StellarGraph-y that isn't already explained in `loading-pandas.ipynb`. I agree that the exact reduction is specific to their data and their use case, but I feel that "filter the elements (in isolation)" and "k-hop subgraphs" are likely to be the most common (or at least, serve as close-enough examples) and so we can spell them out to get people started, and reduce the barriers to working with StellarGraph.

Reply:
> This ~whole notebook is a cypher tutorial, since it's actually not doing anything StellarGraph-y that isn't already explained in loading-pandas.ipynb.

I guess so, but there's some relationship between the query being run and the way the stellargraph constructor is used that the notebook is demonstrating (e.g. run a query to obtain heterogeneous graph then use it to create a heterogeneous stellargraph), whereas for the subgraph one feels more like it's purely in cypher (i.e. stellargraph doesn't know/care that it's a subgraph). That said, I'm fine with keeping it since I agree it's likely to be a common use-case.

Reply:
Ah, I understand the distinction you were seeing, and it makes sense to me now. I'll leave it as is, but we can always rethink.